Documentation/INTERNALS.md

```diff
@@ -451,3 +451,94 @@ TODO
 
 TODO
 
+# Releasing
+
+When doing a release:
+
+1. Examine what has changed
+
+   Githubs compre UI does a reasonable job here.  [Open that UI](https://github.com/apple/swift-protobuf/compare)
+   and set the _base_ to be the previous tag (_vX.Y.Z_), and the _compare_ can be left at _master_
+   since that is what the release is cut off of.
+
+   It usually works best to open the links for each commit you want to look at in a new browser
+   window/tab.  That way you can review each independently rather then looking at the unified
+   diffs.
+
+   When looking at a individual commit, at the top github will show that it was a commit on master
+   and include a reference '#XYZ', this tells you what pull request it was part of.  This is useful
+   for doing the release notes in the next section.
+
+1. Validate the versions numbers
+
+   1. Inspect `Sources/SwiftProtobuf/Version.swift` and ensure the number is what you expect for
+      the next release.
+
+      Normally we try to bump _master_ to a new revision after each release, so this number may
```

Comment:
I'm not following.

github's "releases" are basically a tag with some web ui and the auto creation of tar/zip files of the bits for folks to download.

There are two other patterns I've seen for this: 

- Use even revisions for the _pending_ state and odd for the _released_ state.
- Use zero as the master branch, and all release have a non zero revision.

Both allow you to always tell if some was using `master` directly or not.  But both add complexity in how you cut a release.

Thoughts?